packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -70,9 +100,79 @@ window.\$hotReloadHook = function(modules) {
 /// Generate a synthetic main module which captures the application's main
 /// method.
 String generateMainModule({@required String entrypoint}) {
-  return '''
+  return '''/* ENTRYPOINT_EXTENTION_MARKER */
+// baseUrlScript
+var baseUrl = (function () {
+  // Attempt to detect --precompiled mode for tests, and set the base url
+  // appropriately, otherwise set it to '/'.
+  var pathParts = location.pathname.split("/");
+  if (pathParts[0] == "") {
+    pathParts.shift();
+  }
+  if (pathParts.length > 1 && pathParts[1] == "test") {
+    return "/" + pathParts.slice(0, 2).join("/") + "/";
+  }
+  // Attempt to detect base url using <base href> html tag
+  // base href should start and end with "/"
+  if (typeof document !== 'undefined') {
+    var el = document.getElementsByTagName('base');
+    if (el && el[0] && el[0].getAttribute("href") && el[0].getAttribute
+    ("href").startsWith("/") && el[0].getAttribute("href").endsWith("/")){
+      return el[0].getAttribute("href");
+    }
+  }
+  // return default value
+  return "/";
+}());
+$_currentDirectoryScript
+// dart loader
+if(!window.\$dartLoader) {
+   window.\$dartLoader = {
+     appDigests: _currentDirectory + 'basic.digests',
+     moduleIdToUrl: new Map(),
+     urlToModuleId: new Map(),
+     rootDirectories: new Array(),
+     // Used in package:build_runner/src/server/build_updates_client/hot_reload_client.dart
+     moduleParentsGraph: new Map(),
+     moduleLoadingErrorCallbacks: new Map(),
+     forceLoadModule: function (moduleName, callback, onError) {
+       if (typeof onError != 'undefined') {
+         var errorCallbacks = \$dartLoader.moduleLoadingErrorCallbacks;
+         if (!errorCallbacks.has(moduleName)) {
+           errorCallbacks.set(moduleName, new Set());
+         }
+         errorCallbacks.get(moduleName).add(onError);
+       }
+       requirejs.undef(moduleName);
+       requirejs([moduleName], function() {
+         if (typeof onError != 'undefined') {
+           errorCallbacks.get(moduleName).delete(onError);
+         }
+         if (typeof callback != 'undefined') {
+           callback();
+         }
+       });
+     },
+     getModuleLibraries: null, // set up by _initializeTools
+   };
+}
+let modulePaths = {};
+let customModulePaths = {};
```

Comment:
This variable seems to be populated but never used.

Reply:
See above